src/HydraAuction/Tx/Common.hs

```diff
@@ -121,7 +131,9 @@ currentAuctionStage terms = do
   let matchingStages = filter (member currentTime . stageToInterval terms) auctionStages
   return $ case matchingStages of
     [stage] -> stage
-    _ -> error "Impossible happend: more than one matching stage"
+    [_, stage] -> stage
```

Comment:
Why do we need this?

Reply:
Or maybe this is just wrong and intervals should be ensured to be not intersecting? I missed that.

Reply:
Simple fix breaks Tx construction, cuz they use Upper bounds too, so I added that task into: https://github.com/mlabs-haskell/hydra-auction/issues/268